### PR TITLE
[bitnami/sonarqube] Modify README

### PR DESCRIPTION
**Pull request by ahmedramzygi**
### Description of the change

1) **READMDE.md** file  I have changed the README.md of the sonarqube chart inside the PostgreSQL section to update all the attributes by placing primary after the PostgreSQL. For example, it looks like this when you want to set the existing claim attribute  --> `--set postgresql.primary.persistence.existingClaim=sonar-pv-claim` instead of `postgresql.persistence. existing claim =sonar-pv-claim`.
As if you typed in the set parameter `postgresql.persistence.existingClaim` without accessing the primary attribute it will always create a new PV claim even though you have specified an existing one.
This section of code inside the PostgreSQL chart is updated but the documentation of it inside sonarqube is not updated
2) **values.yaml** file Added the primary section to include all the PostgreSQL attributes inside one of the database types.
    2.1) Also I updated the metadata and comments for the newly introduced attributes from the PostgreSQL bitnami chart 
          documentation
3) **charts.yaml** file Bumped the chart version.

### Benefits

It will prevent the misuse of `--set` of the helm chart of sonarqube using the PostgreSQL section.

### Possible drawbacks

No possible drawbacks.

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/sonarqube/README.md [hunk 1]
```diff
@@ -310,18 +310,18 @@ The command removes all the Kubernetes components associated with the chart and
...
-| `postgresql.persistence.size`          | Size of the created PVCs                                                           | `8Gi`               |
+| `postgresql.primary.enabled`             | Deploy PostgreSQL subchart                                                         | `true`              |
+| `postgresql.primary.nameOverride`        | Override name of the PostgreSQL chart                                              | `""`                |
+| `postgresql.primary.auth.existingSecret` | Existing secret containing the password of the PostgreSQL chart                    | `""`                |
+| `postgresql.primary.auth.password`       | Password for the postgres user of the PostgreSQL chart (auto-generated if not set) | `""`                |
+| `postgresql.primary.auth.username`       | Username to create when deploying the PostgreSQL chart                             | `bn_sonarqube`      |
+| `postgresql.primary.auth.database`       | Database to create when deploying the PostgreSQL chart                             | `bitnami_sonarqube` |
```
Comment (Mauraza) [inline bitnami/sonarqube/README.md:318]: These variables need not be changed if soy check the chart of postgresql does not appear below primary

File: bitnami/sonarqube/README.md [hunk 2]
```diff
@@ -310,18 +310,18 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `postgresql.primary.service.ports.postgresql`  | PostgreSQL service port                                                            | `5432`              |
+| `postgresql.primary.persistence.enabled`       | Use PVCs when deploying the PostgreSQL chart                                       | `true`              |
+| `postgresql.primary.persistence.existingClaim` | Use an existing PVC when deploying the PostgreSQL chart                            | `""`                |
+| `postgresql.primary.persistence.storageClass`  | storageClass of the created PVCs                                                   | `""`                |
+| `postgresql.primary.persistence.accessMode`    | Access mode of the created PVCs                                                    | `ReadWriteOnce`     |
+| `postgresql.primary.persistence.size`          | Size of the created PVCs                                                           | `8Gi`               |
```
Comment (Mauraza) [inline bitnami/sonarqube/README.md:324]: If you want to update these variables, you need to update the [values.yaml](https://github.com/bitnami/charts/blob/main/bitnami/sonarqube/values.yaml) and follow the checklist that appears when you create the PR.

**Review by Mauraza — CHANGES_REQUESTED** [review]
Hi @ahmedramzygi,

Thank you for your contribution! Could you be so kind as to check my comments?

**Comment by Mauraza** [comment]
Hi @ahmedramzygi,

Could you please follow the checklist and update the version of the chart?

File: bitnami/sonarqube/values.yaml [hunk 3]
```diff
@@ -900,28 +900,24 @@ postgresql:
...
-  ## @param primary.name Name of the primary database (eg primary, master, leader, ...)
-  ##
-  primary:
   ## PostgreSQL service
   ##
   service:
-  ## @param postgresql.primary.service.ports.postgresql PostgreSQL service port
-  ##
+    ## @param postgresql.service.ports.postgresql PostgreSQL service port
+    ##
```
Comment (Mauraza) [inline bitnami/sonarqube/values.yaml:907]: The tabulations are needed
```
  ## @param primary.name Name of the primary database (eg primary, master, leader, ...)
  ##
  primary:
    ## PostgreSQL service
    ##
    service:
    ## @param postgresql.primary.service.ports.postgresql PostgreSQL service port
    ##
.....
```
Could you add it in the rest of the values that have primary?

**Review by Mauraza — CHANGES_REQUESTED** [review]

**Comment by ahmedramzygi** [comment]
@Mauraza i have made the requested changes, Thanks for your support.

**Review by Mauraza — CHANGES_REQUESTED** [review]
Hi @ahmedramzygi,

We have almost done it, The part of primary should be in part of Postgresql, Could you check the comments, please?

**Comment by ahmedramzygi** [comment]
Hi @Mauraza when I update the requested changes for the metadata for the postgresql.primary attributes, the pipeline for README.md generator failes of the following error.
image
As It can't get the attribute postgresql.primary.persistence.enabled.

**Comment by Mauraza** [comment]
> Hi @Mauraza when I update the requested changes for the metadata for the postgresql.primary attributes, the pipeline for README.md generator failes of the following error. image As It can't get the attribute postgresql.primary.persistence.enabled.

primary attribute belongs to postgresql, so it has to be tabulated so that `postgres.primary` can recognize it.
As appears in https://github.com/bitnami/charts/pull/14376#discussion_r1083726561

**Comment by ahmedramzygi** [comment]
@Mauraza I have applied the changes requested, Thanks again for your support.

File: bitnami/sonarqube/values.yaml [hunk 4]
```diff
@@ -900,38 +900,39 @@ postgresql:
     ## @param postgresql.auth.database Database to create when deploying the PostgreSQL chart
     ##
     database: bitnami_sonarqube
-  ## PostgreSQL service
+  ## @section PostgreSQL Primary parameters
```
Comment (Mauraza) [inline bitnami/sonarqube/values.yaml:903]: Please remove the `@section` of this line, it is part of the postgresql.

**Review by Mauraza — COMMENTED** [review]
Hi @ahmedramzygi,

The last small change and we end up

File: bitnami/sonarqube/values.yaml (hunk 4 above)
Comment (ahmedramzygi) [reply bitnami/sonarqube/values.yaml]: Done

**Review by Mauraza — APPROVED** [review]
Hi @ahmedramzygi,

Thanks for the contribution!! LGTM!

**Pull request merged**